components/view/report-form.tsx

```diff
@@ -0,0 +1,115 @@
+            toast.error(message);
+            return;
+        }
+
+        toast.success("Report submitted successfully");
+        setIsDialogOpen(false);
+    };
+
+
+    return (
+        <Dialog open={isDialogOpen} onOpenChange={setIsDialogOpen}>
+            <DialogTrigger asChild>
+                <Button
+                    className="m-1 bg-gray-900 text-white hover:bg-gray-900/80"
+                    size="icon"
+                    title="Report abuse"
+                >
+                    <Flag className="h-5 w-5" />
+                </Button>
+            </DialogTrigger>
+            <DialogContent className="sm:max-w-[425px]">
+                <DialogHeader>
+                    <DialogTitle>Report an Issue</DialogTitle>
+                    <DialogDescription>
+                        What kind of issue would you like to report? Please select the most appropriate option.
+                    </DialogDescription>
+                </DialogHeader>
+                <div className="grid gap-4 py-4">
+                    <RadioGroup onValueChange={setAbuseType} className="grid gap-2">
+                        <div className="flex items-center space-x-2">
+                            <RadioGroupItem value="spam" id="spam" />
+                            <Label htmlFor="spam">Spam, Fraud, or Scam</Label>
+                        </div>
+                        <div className="flex items-center space-x-2">
+                            <RadioGroupItem value="malware" id="malware" />
+                            <Label htmlFor="malware">Malware or virus</Label>
+                        </div>
+                        <div className="flex items-center space-x-2">
+                            <RadioGroupItem value="copyright" id="copyright" />
+                            <Label htmlFor="copyright">Copyright violation</Label>
+                        </div>
+                        <div className="flex items-center space-x-2">
+                            <RadioGroupItem value="harmful" id="harmful" />
+                            <Label htmlFor="harmful">Harmful content</Label>
+                        </div>
+                        <div className="flex items-center space-x-2">
+                            <RadioGroupItem value="not-working" id="not-working" />
+                            <Label htmlFor="not-working">Content is not working properly</Label>
+                        </div>
+                        <div className="flex items-center space-x-2">
+                            <RadioGroupItem value="other" id="other" />
+                            <Label htmlFor="other">Other</Label>
+                        </div>
+                    </RadioGroup>
+                </div>
+                <DialogFooter>
+                    <Button onClick={handleSubmit} disabled={!abuseType}>Submit Report</Button>
+                </DialogFooter>
+            </DialogContent>
+        </Dialog>
```

Comment:
Instead of a Dialog, let's make it a Popover, it feels more clean in this case 

https://ui.shadcn.com/docs/components/popover